yoti-sdk-impl/src/main/java/com/yoti/api/client/qrcode/policy/SimpleDynamicPolicyBuilder.java

```diff
@@ -0,0 +1,212 @@
+    private final Set<Integer> wantedAuthTypes = new HashSet<>();
+    private boolean wantedRememberMe;
+    private boolean wantedRememberMeOptional;
+
+    @Override
+    protected DynamicPolicyBuilder createDynamicPolicyBuilder() {
+        return new SimpleDynamicPolicyBuilder();
+    }
+
+    @Override
+    public DynamicPolicyBuilder withWantedAttribute(WantedAttribute wantedAttribute) {
+        String key = wantedAttribute.getDerivation() != null ? wantedAttribute.getDerivation() : wantedAttribute.getName();
+        this.wantedAttributes.put(key, wantedAttribute);
+        return this;
+    }
+
+    private DynamicPolicyBuilder withWantedAttribute(boolean optional, String name) {
+        WantedAttribute wantedAttribute = new SimpleWantedAttributeBuilder()
+                .withName(name)
+                .withOptional(optional)
+                .build();
+        return withWantedAttribute(wantedAttribute);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withFamilyName() {
+        return withFamilyName(false);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withFamilyName(boolean optional) {
+        return withWantedAttribute(optional, AttributeConstants.HumanProfileAttributes.FAMILY_NAME);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withGivenNames() {
+        return withGivenNames(false);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withGivenNames(boolean optional) {
+        return withWantedAttribute(optional, AttributeConstants.HumanProfileAttributes.GIVEN_NAMES);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withFullName() {
+        return withFullName(false);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withFullName(boolean optional) {
+        return withWantedAttribute(optional, AttributeConstants.HumanProfileAttributes.FULL_NAME);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withDateOfBirth() {
+        return withDateOfBirth(false);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withDateOfBirth(boolean optional) {
+        return withWantedAttribute(optional, AttributeConstants.HumanProfileAttributes.DATE_OF_BIRTH);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withAgeOver(int age) {
+        return withAgeOver(false, age);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withAgeOver(boolean optional, int age) {
+        return withAgeDerivedAttribute(optional, AttributeConstants.HumanProfileAttributes.AGE_OVER + age);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withAgeUnder(int age) {
+        return withAgeUnder(false, age);
+    }
+
+    @Override
+    public DynamicPolicyBuilder withAgeUnder(boolean optional, int age) {
+        return withAgeDerivedAttribute(optional, AttributeConstants.HumanProfileAttributes.AGE_UNDER + age);
+    }
+
+    private DynamicPolicyBuilder withAgeDerivedAttribute(boolean optional, String derivation) {
```

Comment:
- trivial - shouldn't be better to club methods by accessibility so this method in the bottom after all the public?

Reply:
That private method is only used by the two public methods above it.  I'd rather keep it grouped with them.